pkg/splunk/controller/serviceaccount.go

```diff
@@ -0,0 +1,57 @@
+// Copyright (c) 2018-2020 Splunk Inc. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// 	http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package controller
+
+import (
+	"context"
+	"reflect"
+
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/types"
+
+	splcommon "github.com/splunk/splunk-operator/pkg/splunk/common"
+	splutil "github.com/splunk/splunk-operator/pkg/splunk/util"
+)
+
+// ApplyServiceAccount creates or updates a Kubernetes serviceAccount
+func ApplyServiceAccount(client splcommon.ControllerClient, serviceAccount *corev1.ServiceAccount) error {
+	namespacedName := types.NamespacedName{Namespace: serviceAccount.GetNamespace(), Name: serviceAccount.GetName()}
+	var current corev1.ServiceAccount
+
+	err := client.Get(context.TODO(), namespacedName, &current)
+	if err == nil {
+		if !reflect.DeepEqual(serviceAccount, &current) {
```

Comment:
Do we need to throw an error or print a log here if the Apply fails?

Reply:
`If err != nil`  it means that the object was found and hence we don't need to error out/print log?

Reply:
Ok so the cases here are: The context doesn't exists and we create it.  Or it exists and we check if its the same and if not, call `UpdateResource`.  If there is an error in either its returned to the caller.  Do we want a log stateing we're updating it?

Reply:
Got it. Making this change as well now.